### DISCUSSION TITLE
What's the alternative to `create_scoped_session` in versions 3+ to create a session for the read replica?

### DISCUSSION THREAD

**Discussion by andrii-korotkov-verkada** [pallets-eco/flask-sqlalchemy#1375] [Q&A] [answered]
Hey. There's a session created for read replica using `SQLAlchemy` object's `create_scoped_session` with `options={"expire_on_commit": False, "bind": database.get_engine(bind="replica")}`. But the method was renamed and marked private in v3. What's the alternative to this? Here's a potentially related question https://github.com/pallets-eco/flask-sqlalchemy/discussions/1374.

**Comment by davidism** [comment, marked as answer]
You can use the private attribute, with the understanding you're using something private that might change. Write tests so you catch if something does change in that case.

Creating a bunch of scoped sessions is probably not doing what you intended it to. It requires you actually accessing the session again in a new scope in order to clean up the old scope. Flask-SQLAlchemy manages this by calling `session.remove()` at request teardown. So you probably just want to create instances of `flask_sqlalchemy.session.Session(db)` instead.

You might want to look at Flask-SQLAlchemy-Lite as an alternative if you want to manage the session yourself anyway: https://flask-sqlalchemy-lite.readthedocs.io/en/latest/session/
